src/manifolds/StiefelFrechet.jl

```diff
@@ -0,0 +1,314 @@
+@doc raw"""
+    q, dot_q = dot_exp(M::MetricManifold{ℝ, Stiefel{n,k,ℝ}, <:StiefelSubmersionMetric}, p, X)
+    dot_exp!(M::MetricManifold{ℝ, Stiefel{n,k,ℝ}, <:StiefelSubmersionMetric}, q, dot_q, p, X)
+
+Compute the geodesic end point ``q = q(t)`` and its time derivative ``dot\_q = \dot{q}(t)`` on [`Stiefel(n,k)`](@ref) with the [`StiefelSubmersionMetric`](@ref). The geodesic is given by
+````math
+\operatorname{Exp} tX = \begin{bmatrix} p & X\end{bmatrix}
+\{\exp t\begin{bmatrix} (2\bar{\alpha}-1)A & 2(\bar{\alpha}-1)A^2-X^{\mathrm{T}}X \\ I_k & A\end{bmatrix}\}
+\begin{bmatrix}\ \exp((1-2\bar{\alpha})tA) \\ 0\end{bmatrix}.
+````
+with ``\bar{\alpha} = \frac{1}{2(1+\alpha)}, A = p^{\mathrm{T}}X``. This implementation is based on [^Nguyen2022], equation (3).
+
+[^Nguyen2022]:
+    ^Nguyen, D. (2022),
+    ^"Closed-form Geodesics and Optimization for Riemannian Logarithms of Stiefel and Flag Manifolds".
+    ^"J Optim Theory Appl 194, 142–166 (2022)."
+    ^doi: https://doi.org/10.1007/s10957-022-02012-3
+    ^url https://rdcu.be/c0V2r
+"""
+function dot_exp(
+    M::MetricManifold{ℝ,Stiefel{n,k,ℝ},<:StiefelSubmersionMetric},
+    p,
+    X,
+    t,
+) where {n,k}
+    q = similar(p)
+    dot_q = similar(p)
```

Comment:
I think `dot_q` is a tangent vector, right? First of all, then `dot_q`should be called `diff_exp` or `differential_exp` but even more, `dot_q` should be initialised as a tangent vector (`zero_vector(M,p)` probably) and better be called `X` or `Y` following our naming convention.

Reply:
OK will do